services/huaweicloud/src/main/java/org/eclipse/osc/orchestrator/plugin/huaweicloud/builders/terraform/TFExecutor.java

```diff
@@ -0,0 +1,139 @@
+package org.eclipse.osc.orchestrator.plugin.huaweicloud.builders.terraform;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.file.FileSystems;
+import java.util.Arrays;
+import java.util.Map;
+import lombok.extern.slf4j.Slf4j;
+import org.eclipse.osc.orchestrator.plugin.huaweicloud.exceptions.TFExecutorException;
+import org.eclipse.osc.services.ocl.loader.Ocl;
+
+@Slf4j
+public class TFExecutor {
+
+    private final Ocl ocl;
+    private final Map<String, String> env;
+    private String tfPath;
+    private String workPath;
+
+    public TFExecutor(final Ocl ocl, Map<String, String> env) {
+        this.ocl = ocl;
+        this.env = env;
+    }
+
+    public void createWorkspace() {
+        File ws =
+            new File("terraform_ws" + FileSystems.getDefault().getSeparator() + ocl.getName());
```

Comment:
when to delete this dir?

Reply:
This will be a temporary directory, valuable information will be extracted to persistent storage, then this dir can be cleared safelly.

Reply:
This should be with the service lifecycle unless the managed serveice destroyed, maybe we can remove this dir with that operation after `terraform destroy`

Reply:
Compared to the long lifecycle of the managed service,
I think the directories are too weak to hold such important things.
Anyway, let us have a further discussion about this.

Reply:
ok, agree. but it's ok for now as a PoC.